### PR TITLE
Bug fix 3.6/restart follower no leader set

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose

Proposal instead of https://github.com/arangodb/arangodb/pull/11667. This code ensures that a restarting server will properly update the follower info for all shards (as those are only in memory).

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *(please describe tests)*.

### CODE REVIEW

File: arangod/Cluster/Maintenance.cpp [hunk 1]
```diff
@@ -298,7 +303,7 @@ void handlePlanShard(VPackSlice const& cprops, VPackSlice const& ldb,
       // Index errors are checked in `compareIndexes`. THe loop below only
       // cares about those indexes that have no error.
       if (difference.slice().isArray()) {
-        for (auto const& index : VPackArrayIterator(difference.slice())) {
+        for (auto const&& index : VPackArrayIterator(difference.slice())) {
```
Comment (neunhoef) [inline arangod/Cluster/Maintenance.cpp:306]: Why is this change good or necessary?

**Comment by joerg84** [comment]
The decision was to move this into the 3.6.5 release

**Comment by maierlars** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr/10103/

**Comment by KVS85** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr/10106/

**Comment by KVS85** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr/10109/

**Comment by KVS85** [comment]
https://jenkins.arangodb.biz/job/arangodb-matrix-pr-mac-onlycatch/7778/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**